tests/lib/rules/no-get-with-default.js

```diff
@@ -17,25 +17,70 @@ ruleTester.run('no-get-with-default', rule, {
     "getWithDefault.testMethod(testClass, 'key', [])",
   ],
   invalid: [
+    // this.getWithDefault
     {
-      code: "const test = this.getWithDefault('key', []);",
+      code: "const test = this.getWithDefault('key', []);", // With a string property.
+      output: "const test = (this.get('key') === undefined ? [] : this.get('key'));",
```

Comment:
For non-chained static string paths we should emit:

```suggestion
      output: "const test = this.key === undefined ? [] : this.key;",
```

Reply:
I considered doing that but realized it would be significantly easier to leave that job for the `no-get` rule to handle. The [no-get](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-get.md) rule already has a lot of logic for determining whether `this.get('key')` can be converted to `this.key` including checking for chained paths, proxies, unknownMethod, etc. And fortunately, these two recommended rules can work together, as the `no-get` rule will immediately run on the result of `no-get-with-default`, meaning that there's no difference for the end user (unless the user happened to disable `no-get`).